### DISCUSSION TITLE
How to consume JustJ in a Tycho build?

### DISCUSSION THREAD

**Discussion by guw** [eclipse-justj/justj#32] [Q&A]
I'm having some trouble getting the target definition to resolve properly.

Here is the error I am struggling with:
```
[ERROR] Cannot resolve target definition:
[ERROR]   Software being installed: a.jre.javase 21.0.0
[ERROR]   Software being installed: org.eclipse.justj.openjdk.hotspot.jre.full.stripped 21.0.5.v20241023-1957
[ERROR]   Cannot satisfy dependency: org.eclipse.justj.openjdk.hotspot.jre.full.stripped 21.0.5.v20241023-1957 conflicts with: org.eclipse.equinox.p2.iu; a.jre.javase 0.0.0, min=0, max=0, filter=(!(org.eclipse.justj.buildtime=true))
[ERROR] 
```

This is Tycho version 4.0.9. I am a bit puzzled by the filter.

The `.target` platform is defined with:
```
<targetJRE path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-21"/>
```

**Comment by merks** [comment]
Did you look at the documentation?

https://eclipse.dev/justj/?page=documentation

😜 

This is what EPP does 

https://github.com/eclipse-packaging/packages/blob/24a435ee959da3b74bc285cb4e0242fe2dd7a10a/releng/org.eclipse.epp.config/parent/pom.xml#L131

**Reply by guw** [reply]
🤦‍♂️ I did not scoll down far enough. Thanks Ed!

**Reply by guw** [reply]
Should I use a different one than `org.eclipse.justj.openjdk.hotspot.jre.full-21` if my `target` file contains only the stripped version?

**Reply by merks** [reply]
Yes, match the name and version you are actually using.  The Repository Explorer is probably helpful...
